### PR TITLE
{Hybridkubernetes} Adding hybridkubernetes mgmt sdk support in _shared.py

### PR DESCRIPTION
**Pull request by alphaWizard**
SDK documentation: https://docs.microsoft.com/en-us/python/api/azure-mgmt-hybridkubernetes/azure.mgmt.hybridkubernetes?view=azure-python
Published package: https://pypi.org/project/azure-mgmt-hybridkubernetes/

Our service went GA on 1st March, 2021. 
Resource Provider Namespace: _**Microsoft.Kubernetes**_
Resource Type: **_Microsoft.Kubernetes/ConnectedClusters_**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
hybridkubernetes

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls review this one?

**Comment by fengzhou-msft** [comment]
@alphaWizard how will this change be used?

**Comment by alphaWizard** [comment]
> @alphaWizard how will this change be used?

@fengzhou-msft , We plan to let other extensions/partners create and use the **ConnectedKubernetesClient** through get_mgmt_service_client. Sample below:
```
from azure.cli.core.commands.client_factory import get_mgmt_service_client
from azure.cli.core.profiles import ResourceType
cc_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_HYBRIDKUBERNETES, subscription_id=subscription_id).connected_cluster
cc_obj = cc_client.get(resource_group_name, cluster_name)
```


Let me know, if that is the correct/suggested way?

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -210,7 +211,8 @@ def default_api_version(self):
         ResourceType.MGMT_APPSERVICE: '2019-08-01',
         ResourceType.MGMT_IOTHUB: '2020-03-01',
         ResourceType.MGMT_ARO: '2020-04-30',
-        ResourceType.MGMT_DATABOXEDGE: '2019-08-01'
+        ResourceType.MGMT_DATABOXEDGE: '2019-08-01',
+        ResourceType.MGMT_HYBRIDKUBERNETES: '2021-03-01'
```
Comment (Juliehzl) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:215]: only support in latest profile?

**Comment by Juliehzl** [comment]
Hi @alphaWizard, I think I need more information for your service. Are hybridkubernetes related commands support in your extension? If yes, you could consider define resource type in your own extension. For how to define resource type in extension, you could refer to:
https://github.com/Azure/azure-cli-extensions/blob/e434fbf47d8eff609d846a79addbb673009cfbef/src/storage-preview/azext_storage_preview/profiles.py#L11-L12

In addition, may I know why you need such resource type? Does your service want to support Azure Stack with multiple api versions?

**Pull request closed without merging**